<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>9. Count reads with HtSeq &mdash; RNA_Seq 0.01 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.01',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://c328740.ssl.cf1.rackcdn.com/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="RNA_Seq 0.01 documentation" href="index.html" />
    <link rel="next" title="10. Processing Next Sample Through Automated Pipeline to get counts" href="batch_submission_for_other_sample.html" />
    <link rel="prev" title="8. Mapping reads with Tophat2" href="mapping_reads_with_tophat.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="batch_submission_for_other_sample.html" title="10. Processing Next Sample Through Automated Pipeline to get counts"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="mapping_reads_with_tophat.html" title="8. Mapping reads with Tophat2"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">RNA_Seq 0.01 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="count-reads-with-htseq">
<h1>9. Count reads with HtSeq<a class="headerlink" href="#count-reads-with-htseq" title="Permalink to this headline">Â¶</a></h1>
<ol class="arabic simple">
<li>Now we need to count the number of reads that mapped to each gene</li>
</ol>
<p>So, load the modules PySAM and HTSeq</p>
<div class="highlight-python"><div class="highlight"><pre>module load PySAM/0.6
module load HTSeq/0.6.1
</pre></div>
</div>
<p>Now run HTSeq to get the counts</p>
<div class="highlight-python"><div class="highlight"><pre>    htseq-count --format=bam --stranded=no --order=pos \
tophat_salivary_repl1/accepted_hits.bam \
~/RNAseq-model/Homo_sapiens/Ensembl/GRCh37/Annotation/Genes/genes.gtf &gt; salivary_repl1_counts.txt
</pre></div>
</div>
<p>Lets see what the output was</p>
<div class="highlight-python"><div class="highlight"><pre>less salivary_repl1_counts.txt
</pre></div>
</div>
<p>So you can see two columns. With the first column the indicates gene name and the second column indicates the counts</p>
<p>use &#8220;q&#8221; to exit</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="mapping_reads_with_tophat.html"
                        title="previous chapter">8. Mapping reads with Tophat2</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="batch_submission_for_other_sample.html"
                        title="next chapter">10. Processing Next Sample Through Automated Pipeline to get counts</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/count_with_htseq.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="batch_submission_for_other_sample.html" title="10. Processing Next Sample Through Automated Pipeline to get counts"
             >next</a> |</li>
        <li class="right" >
          <a href="mapping_reads_with_tophat.html" title="8. Mapping reads with Tophat2"
             >previous</a> |</li>
        <li><a href="index.html">RNA_Seq 0.01 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Sudeep Perumbakkam.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3b2.
    </div>
  </body>
</html>