<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>5. Obtaining Data and Read Quality &mdash; RNA_Seq 0.01 documentation</title>
    
    <link rel="stylesheet" href="_static/default.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.01',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://c328740.ssl.cf1.rackcdn.com/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="top" title="RNA_Seq 0.01 documentation" href="index.html" />
    <link rel="next" title="6. Trimming Fastq Sequences" href="trimming.html" />
    <link rel="prev" title="4. Basic Commands" href="basic_commands.html" /> 
  </head>
  <body role="document">
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="trimming.html" title="6. Trimming Fastq Sequences"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="basic_commands.html" title="4. Basic Commands"
             accesskey="P">previous</a> |</li>
        <li><a href="index.html">RNA_Seq 0.01 documentation</a> &raquo;</li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <div class="section" id="obtaining-data-and-read-quality">
<h1>5. Obtaining Data and Read Quality<a class="headerlink" href="#obtaining-data-and-read-quality" title="Permalink to this headline">¶</a></h1>
<p>The data we will be using has been downloaded from the source. You will not be doing this part.</p>
<p>Two samples: salivary gland and lung. Note that each sample has one replicates, and each replicate has two files.</p>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Since downloading this data takes hours, the data has been already downloaded and put into a folder for connivence</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p>The command line is case sensitive. So be careful when to type commands. The best practice is to use lower case, underscore (_), and no spaces between words eg. course_work_project_mmg433. Also try to be as explicit in naming folders or use a data stamp eg. mmg443_rnaseq_tutorial_1/1/2015.</p>
<ol class="last arabic simple">
<li>Lets start the process by first logging on to the HPCC</li>
</ol>
</div>
<p>Now that you are logged  into the HPC with SSH; use your MSU NetID and log into the machine ‘gateway.hpcc.msu.edu’.</p>
<div class="highlight-python"><div class="highlight"><pre>module load powertools
getexample RNAseq-model
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">This loads the data from the machine to your home folder</p>
</div>
<p>Now log into a node:</p>
<div class="highlight-python"><div class="highlight"><pre>ssh dev-intel14-phi
</pre></div>
</div>
<p>Once you log into the dev node, type or paste the following</p>
<div class="highlight-python"><div class="highlight"><pre>ls -la ~/RNAseq-model/data/
</pre></div>
</div>
<p>This is what you will see (a list of all files with size and owner)</p>
<div class="highlight-python"><div class="highlight"><pre>-r--r--r-- 1 mscholz common-data 3714262571 Dec  4 08:44 ERR315325_1.fastq
-r--r--r-- 1 mscholz common-data 3714262571 Dec  4 08:44 ERR315325_2.fastq
-r--r--r-- 1 mscholz common-data 2365633645 Dec  4 08:44 ERR315326_1.fastq
-r--r--r-- 1 mscholz common-data 2365633645 Dec  4 08:44 ERR315326_2.fastq
</pre></div>
</div>
<p>Now we have to copy this data to your home directory</p>
<ol class="arabic simple" start="2">
<li>Copying the original data to your folder by first making a directory:</li>
</ol>
<div class="highlight-python"><div class="highlight"><pre>mkdir ~/rnaseq
cd ~/rnaseq
</pre></div>
</div>
<p>Now we are not going to work with the complete dataset, just a small fraction (50,000 reads):</p>
<div class="highlight-python"><div class="highlight"><pre>head -50000 ~/RNAseq-model/data/ERR315325_1.fastq | gzip &gt; salivary_repl1_R1.fq.gz
head -50000 ~/RNAseq-model/data/ERR315325_2.fastq | gzip &gt; salivary_repl1_R2.fq.gz
</pre></div>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">What are above commands doing? First of all there are 2 programs being used in this process. We call head, a program which reads the top of lines of a file (1 to 10000 etc.) to read the first 50000 lines from the file ERR315325_1.fastq. We are also giving what is called the &#8220;complete path&#8221; (~/RNAseq-model/data/ERR315325_1.fastq), which is a &#8220;good practice&#8221; when using the command line. Now the 200000 lines from the fastq file is transfered to gzip, another program, in the pipeline. The symbol &#8220;|&#8221; is called &#8220;piped&#8221;, which takes the output from one program, in our case head, and inputs to the other program, gzip. So gzip now takes the 200000 lines and renames it salivary_repl1_R1.fq.gz and then zips it to reduce space.</p>
</div>
<p>Now we have two files, with R1 and R2 suffix.</p>
<ol class="arabic simple" start="3">
<li>What is a fastq file?</li>
</ol>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">You do not want to open fastq files using word, excel, other programs due to corruption issues. The best is to use the terminal to open the files these big.</p>
</div>
<div class="admonition note">
<p class="first admonition-title">Note</p>
<p class="last">Why do you have two files for each sample?</p>
</div>
<p>info: <a class="reference external" href="http://en.wikipedia.org/wiki/FASTQ_format">http://en.wikipedia.org/wiki/FASTQ_format</a></p>
<img alt="_images/fastq_fig.png" src="_images/fastq_fig.png" />
<ol class="arabic simple" start="4">
<li>Examine the fastq file</li>
</ol>
<div class="highlight-python"><div class="highlight"><pre>less salivary_repl1_R1.fq.gz
</pre></div>
</div>
<p>You can see the file organization</p>
<p>hit &#8220;q&#8221; to quit from the screen and go back to promt.</p>
<ol class="arabic simple" start="4">
<li>Check quality of fastq file</li>
</ol>
<div class="highlight-python"><div class="highlight"><pre>module load FastQC/0.11.2
</pre></div>
</div>
<ul class="simple">
<li>Now run FASTQC on both the files</li>
</ul>
<div class="highlight-python"><div class="highlight"><pre>fastqc salivary_repl1_R1.fq.gz
fastqc salivary_repl1_R2.fq.gz
</pre></div>
</div>
<ul class="simple">
<li>Now we can view the results of the analysis</li>
<li>The files that were generated with the fastqc run (run ls):</li>
</ul>
<div class="highlight-python"><div class="highlight"><pre><span class="n">ls</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="n">salivary_repl1_R1_fastqc</span><span class="o">.</span><span class="n">html</span>
<span class="n">salivary_repl1_R1_fastqc</span><span class="o">.</span><span class="n">zip</span>
<span class="n">salivary_repl1_R2_fastqc</span><span class="o">.</span><span class="n">html</span>
<span class="n">salivary_repl1_R2_fastqc</span><span class="o">.</span><span class="n">zip</span>
</pre></div>
</div>
<p>Copy these files to your laptop. Now use Filezilla to log into your account then drag and drop to your computer and open using a bowser.</p>
</div>


          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <h4>Previous topic</h4>
  <p class="topless"><a href="basic_commands.html"
                        title="previous chapter">4. Basic Commands</a></p>
  <h4>Next topic</h4>
  <p class="topless"><a href="trimming.html"
                        title="next chapter">6. Trimming Fastq Sequences</a></p>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/getdata_readquality.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    <p class="searchtip" style="font-size: 90%">
    Enter search terms or a module, class or function name.
    </p>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="trimming.html" title="6. Trimming Fastq Sequences"
             >next</a> |</li>
        <li class="right" >
          <a href="basic_commands.html" title="4. Basic Commands"
             >previous</a> |</li>
        <li><a href="index.html">RNA_Seq 0.01 documentation</a> &raquo;</li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &copy; Copyright 2015, Sudeep Perumbakkam.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.3b2.
    </div>
  </body>
</html>